src/tss2-fapi/api/Fapi_GetEsysBlob.c

```diff
@@ -0,0 +1,388 @@
+ *
+ * Deletes a given key, policy or NV index from the system.
+
+ * Call Fapi_GetEsysBlob_Finish to finish the execution of this command.
+ *
+ * @param[in,out] context The ESAPI_CONTEXT
+ * @param[in] path The path of the FAPI object.
+ *
+ * @retval TSS2_RC_SUCCESS: if the function call was a success.
+ * @retval TSS2_FAPI_RC_BAD_REFERENCE: if context or path is NULL.
+ * @retval TSS2_FAPI_RC_BAD_CONTEXT: if context corruption is detected.
+ * @retval TSS2_FAPI_RC_BAD_PATH: if path does not map to a FAPI entity.
+ * @retval TSS2_FAPI_RC_NOT_DELETABLE: if the entity is not deletable or the
+ *         path is read-only.
+ * @retval TSS2_FAPI_RC_BAD_SEQUENCE: if the context has an asynchronous
+ *         operation already pending.
+ * @retval TSS2_FAPI_RC_IO_ERROR: if the data cannot be saved.
+ * @retval TSS2_FAPI_RC_MEMORY: if the FAPI cannot allocate enough memory for
+ *         internal operations or return parameters.
+ * @retval TSS2_FAPI_RC_NO_TPM if FAPI was initialized in no-TPM-mode via its
+ *         config file.
+ * @retval TSS2_FAPI_RC_PATH_NOT_FOUND if a FAPI object path was not found
+ *         during authorization.
+ * @retval TSS2_FAPI_RC_KEY_NOT_FOUND if a key was not found.
+ * @retval TSS2_FAPI_RC_BAD_VALUE if an invalid value was passed into
+ *         the function.
+ * @retval TSS2_ESYS_RC_* possible error codes of ESAPI.
+ */
+TSS2_RC
+Fapi_GetEsysBlob_Async(
+    FAPI_CONTEXT   *context,
+    char     const *path)
+{
+    LOG_TRACE("called for context:%p", context);
+    LOG_TRACE("path: %s", path);
+
+    TSS2_RC r;
+
+    /* Check for NULL parameters */
+    check_not_null(context);
+    check_not_null(path);
+
+    /* Helpful alias pointers */
+    IFAPI_GetEsysBlob * command = &(context->cmd.GetEsysBlob);
+    IFAPI_OBJECT *object = &command->object;
+    IFAPI_OBJECT *authObject = &command->auth_object;
+
+    /* Copy parameters to context for use during _Finish. */
+    strdup_check(command->path, path, r, error_cleanup);
+
+    object->objectType = IFAPI_OBJ_NONE;
+    authObject->objectType = IFAPI_OBJ_NONE;
+
+    /* Check whether TCTI and ESYS are initialized */
+    return_if_null(context->esys, "Command can't be executed in none TPM mode.",
+                   TSS2_FAPI_RC_NO_TPM);
+
+    /* If the async state automata of FAPI shall be tested, then we must not set
+       the timeouts of ESYS to blocking mode.
+       During testing, the mssim tcti will ensure multiple re-invocations.
+       Usually however the synchronous invocations of FAPI shall instruct ESYS
+       to block until a result is available. */
+#ifndef TEST_FAPI_ASYNC
+    r = Esys_SetTimeout(context->esys, TSS2_TCTI_TIMEOUT_BLOCK);
+    return_if_error_reset_state(r, "Set Timeout to blocking");
+#endif /* TEST_FAPI_ASYNC */
+
+    /* A TPM session will be created to enable object authorization */
+    r = ifapi_session_init(context);
+    return_if_error(r, "Initialize GetEsysBlob");
+
+    r = ifapi_get_sessions_async(context,
```

Comment:
Where is the corresponding `ifapi_get_sessions_finish()` or is the async not even needed ?

Reply:
Yes, the async call is not needed. The session is created in ifapi_load_key.